README.md

```diff
@@ -1,6 +1,93 @@
+
+Inversely, each epoch number within a logical Look instance corresponds to one atomic update in the lifetime of this Look; most often an update to the data it stores, although system-level updates, such as changes to data schema, its processing logic, or deployment topology, also get their own unique epoch numbers, interleaved with data updates in a totally ordered fashion.
+
+### In-Memory Event Store
+
+Under the hood, Look maintains a complete event log of all the mutations that took place, both to the data and to the logic that handles this data. Thus, it is accurate to say that Look is based on an immutable, append-only event log.
+
+The in-memory storage of Look follows the approach of [persistent data structures](https://en.wikipedia.org/wiki/Persistent_data_structure), the nodes of which are compactified and serialized to disk. The storage keeps the "dirtied" pages as well, on disk for the history, as well as in memory, for a configurable amount of wall time and epoch number; some 15 seconds and a million mutations by default.
+
+To a certain degree, this implies that read queries that refer to a specific epoch number are fully idempotent, as long as they are performed against a replica that is configured to store enough of the past state. It is also possible to spin a dedicated replica that is configured to keep more of historic data in hot storage. The storage layer of Look also performs regular snapshots, so, for reconciliation reasons, it is possible to "roll it back", temporarily and on a dedicated node or set of nodes, to the state of data & code that was actual far away in the past.
+
+Last but not least: adding Look replicas is a straightforward task. For read-only replicas, it is as trivial as spinning up a node and pointing it at the currently up and running Look cluster. For read-write replicas, a.k.a. potential leaders when it comes to the high availability and durability of the distributed system, the node is first added as read-only, and, as it has caught up, the record that registers it as such can be added into the cluster configuration namespace of the immutable append-only log of this Look deployment.
+
+### High-Throughput Distributed Consensus
+
+Extra effort is required to keep the system highly available and consistent in an environment where the nodes and/or the network links between them can fail. The throughput that Load can handle is far beyond what modern-day most mature leader elections solutions such as `etcd` can offer.
+
+The solution is to build a custom high-throughput distributed consensus engine by leveraging a low-throughput one. What is elected, say, once every few seconds, is not the leader for the storage engine, but the leader for the front-facing gateway that is responsible for batching the stream of incoming requests so that the actual mutations take place not more than a ~50 times per second.
```

Comment:
Hm, I remember that something was said about principal unwilling to deal with any Consensus Algo implementation because of huge risk of fail. But here it's suppsed to build ourself. Do you assume that using a leveraged one will insure us against probable collapse?

Reply:
Yes, ideally we don't need to build any, but end up leveraging one (or several) existing options, such as `etcd`. This part alone may well be very interesting to talk about.

(And this is what I keep referring to as the Total Order Broadcast Engine), but by itself it's a bit too abstract to make enough people interested =)